src/include/crowbar/complex.rb

```diff
@@ -695,12 +617,195 @@ def StoreRepoURL(key, event)
+            elsif url.include? "ks/dist/child"
+              @repos_location = "sm"
+              @remote_server_url = url.gsub(/(^.*)\/ks\/dist\/child\/(.*)/,"\\1")
+            end
+          end
+        end
+      end
+
+      if @repos_location != "custom"
+        UI.ReplaceWidget(
+          Id("repos_rp"),
+          VBox(
+            VSpacing(0.4),
+            InputField(
+              Id("repos_location_url"),
+              Opt(:hstretch, :notify),
+              # text entry label
+              _("Server &URL")
+            ),
+            VStretch()
+          )
+        )
+      else
+        UI.ReplaceWidget(
+          Id("repos_rp"),
+          VBox(
+            VSpacing(0.4),
+            Table(
+              Id("repos_table"),
+              Opt(:notify, :immediate, :hstretch),
+              Header(
+                # table header
+                _("Repository Name"),
+                _("URL"),
+                _("Ask On Error"),
+                _("Target Platform")
+              )
+            ),
+            # checkbox label
+            Left(CheckBox(Id("ask_on_error"), Opt(:notify), _("&Ask On Error"))),
+            VSpacing(),
+            InputField(
+              Id("repo_url"),
+              Opt(:notify, :hstretch),
+              # text entry label
+              _("Repository &URL")
+            ),
+            # label (hint for user)
+            Left(Label(_("Empty URL means that default value will be used."))),
+            VSpacing(0.4),
+            Left(RadioButtonGroup(
+              Id("target_platform"),
+              # frame label
+              Frame(_("Target Platform"), HBox(
+                HSpacing(),
+                VBox(
+                  Left(RadioButton(Id("common"), Opt(:notify), @platform2label["common"] || "")),
+                  Left(RadioButton(Id("suse-11.3"), Opt(:notify), @platform2label["suse-11.3"] || "")),
+                  Left(RadioButton(Id("suse-12.0"), Opt(:notify), @platform2label["suse-12.0"] || ""))
+                )
+              ))
+            )),
+            VSpacing(),
+            # push button label
+            Left(PushButton(Id("add_repository"), _("A&dd Repository")))
+          )
+        )
+      end
+
+      # initialization of ReplacePoint content
+      if @repos_location == "custom"
+        InitReposTable("repos_table")
+        HandleReposTable("repos_table", { "force" => true })
+      elsif @repos_location != ""
+        InitLocationURL("repos_location_url")
+      end 
+      nil
+    end
+
+    def HandleRPRepos(key, event)
+      subkey = event["ID"]
+
+      case subkey
+      when "repos_table"
+        HandleReposTable(subkey, event)
+      when "repo_url"
+        HandleRepoURL(subkey, event)
+      when "ask_on_error"
+        HandleAskOnError(subkey, event)
+      when "repos_location_url"
+        HandleLocationURL(subkey, event)
+      when "add_repository"
+        HandleAddRepositoryButton(subkey, event)
```

Comment:
Hard to say. For example some events are additionally handled after the case.
And I might ignore some events intentionally, because they are already handled in a different way (this whole CWM setup is mess)